Lib/test/test_subprocess.py

```diff
@@ -1485,6 +1485,50 @@ def test_exception_bad_args_0(self):
         else:
             self.fail("Expected OSError: %s" % desired_exception)
 
+    # We mock the __del__ method for Popen in the next two tests
+    # because it does cleanup based on the pid returned by fork_exec
+    # along with issuing a resource warning if it still exists. Since
+    # we don't actually spawn a process in these tests we can forego
+    # the destructor. An alternative would be to set _child_created to
+    # False before the destructor is called but there is no easy way
+    # to do that
+    @mock.patch.object(subprocess.Popen, "__del__")
+    @mock.patch("subprocess._posixsubprocess.fork_exec")
+    def test_exception_errpipe_normal(self, fork_exec, destructor):
+        """Test error passing done through errpipe_write in the good case"""
+        def proper_error(*args):
+            errpipe_write = args[13]
+            # 15 is the unix error code for EISDIR: 'is a directory'
+            os.write(errpipe_write, b"OSError:15:")
+            return 0
+
+        fork_exec.side_effect = proper_error
+
+        with self.assertRaises(IsADirectoryError):
+            subprocess.Popen(["cmd"])
+
+    @mock.patch.object(subprocess.Popen, "__del__")
+    @mock.patch("subprocess._posixsubprocess.fork_exec")
+    def test_exception_errpipe_bad_data(self, fork_exec, destructor):
+        """Test error passing done through errpipe_write where its not
+        in the expected format"""
+        error_data = b"\xFF\x00\xDE\xAD"
```

Comment:
The reason I chose this is because it is almost certainly not valid in any text encoding. So if anyone accidentally attempts to decode data in the bad path, this will start failing. Why do you suggest using a simple ASCII string?